stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -238,12 +404,20 @@ class cholesky_scalar : public vari {
 inline Eigen::Matrix<var, -1, -1> cholesky_decompose(
     const Eigen::Matrix<var, -1, -1>& A) {
   check_square("cholesky_decompose", "A", A);
-  check_symmetric("cholesky_decompose", "A", A);
-
   Eigen::Matrix<double, -1, -1> L_A(value_of_rec(A));
+#ifdef STAN_OPENCL
+  if (L_A.rows() > opencl_context.tuning_opts().cholesky_size_worth_transfer) {
```

Comment:
The reasons why we left it here is due to the comment above that states: 

```
Internally calls Eigen::LLT rather than using stan::math::cholesky_decompose in order to use an inplace decomposition.
```
But on second thought, if someone defines STAN_OPENCL then the CPU version would only be called for smaller matrices where it doesnt really matter if we do them inplace or not.